plugins/clipboard/plugin.js

```diff
@@ -1222,11 +1222,14 @@
 		}
 
 		function stateFromNamedCommand( command ) {
-			if ( inReadOnly && command in { paste: 1, cut: 1 } )
+			// We need to correctly update toolbar states on readOnly (#2775).
+			if ( ( inReadOnly || editor.readOnly ) && command in { paste: 1, cut: 1 } ) {
```

Comment:
Won't it be better to ensure that `inReadOnly` variable contains correct info?

Reply:
This variable is used only here, but it is updated in other places, I'll move it here.